Support PHP 8.5

sukhwinder33445 opened the pull request:
### PHP 8.4 changes:
- [Function parameters that are null by default must be declared nullable](https://www.php.net/manual/en/migration84.deprecated.php#migration84.deprecated.core).

### Other changes:
- Remove unused patches (the patched library is no longer in use).
- Cleanup phpstan baseline.

resolves #78

sukhwinder33445 committed:
PHP 8.4: Change implicit nullable type declaration to explicit

Since PHP 8.4 implicitly nullable parameter types are deprecated.

sukhwinder33445 committed:
Cleanup phpstan baseline

BastianLedererIcinga reviewed (commented):
The changes look good, everything should be fine once the issue with the dependencies is resolved.

sukhwinder33445 committed:
Cleanup leftovers from dependency replacement

bc4d059 replaced dependencies but left outdated code, addressed here:

- Catch `Throwable` instead of removed `WebSocket\ConnectionException`
- Remove patch for replaced library

sukhwinder33445 commented:
`karriere/pdf-merge:^3.4` does not support `PHP 8.2 and 8.5` simultaneously, which leads to [failing](https://github.com/Icinga/icingaweb2-module-pdfexport/actions/runs/20964804709/job/60252009229?pr=79) tests.

We could consider replacing it with [`sinanbekar/pdf-merger`](https://packagist.org/packages/sinanbekar/pdf-merger). However, currently PDFs cannot be generated because support for [`wikimedia/less.php version 5.4.0`](https://github.com/Icinga/icingaweb2/pull/5457) is still missing. I will test it again as soon as it is ready.

lippserd reviewed (changes requested):
Please merge the two commits that are solely for cleaning up the dependency replacement made in #77. And please rephrase the commit message and description to clarify the intent:

```
Cleanup leftovers from dependency replacement

bc4d059 replaced dependencies but left outdated code, addressed here:

- Catch `Throwable` instead of removed `WebSocket\ConnectionException`
- Remove patch for replaced library
```

Please always use short commit hashes in commit descriptions instead of GitHub PR numbers, as these have no value outside of GitHub. And please do not use bullet points if there is only one point.

Please also adjust the PR description. The only changes required to support PHP 8.5 are now the fix for implicit nullable type declarations, right? I'm also not entirely sure if it makes sense to require the missing react/child-process issue, as it requires a PR in icinga-php-thirdparty that has not yet been merged. So why is the issue resolved in the first place? If this is really a strict requirement, please require the icinga-php-thirdparty PR or remove it.
